src/js/editing/Typing.js

```diff
@@ -45,14 +46,23 @@ define([
       var nextPara;
       // on paragraph: split paragraph
       if (splitRoot) {
-        nextPara = dom.splitTree(splitRoot, rng.getStartPoint());
+        // if new line has content (not a line break)
+        if (!/^BR/i.test(splitRoot.firstChild.nodeName)) {
```

Comment:
I's better to change this condition to `!dom.isEmpty`
https://github.com/summernote/summernote/blob/develop/src/js/core/dom.js#L268